### PR TITLE
Move UfsManager to CoreMasterContext

### PR DESCRIPTION
**Pull request by apc999**
Fix https://github.com/Alluxio/alluxio/issues/9240

This moves provides `UfsManager` to master context so it can be shared across different masters. In this way, `DefaultMetaMaster` no longer needs to call UFS Factory to make UFS instances but consolidating to `UfsManager`

### CODE REVIEW

File: core/server/master/src/main/java/alluxio/master/meta/DefaultMetaMaster.java [hunk 1]
```diff
@@ -243,13 +246,8 @@ public void resetState() {
...
 
-    if (URIUtils.isLocalFilesystem(ServerConfiguration
-        .get(PropertyKey.MASTER_MOUNT_TABLE_ROOT_UFS))) {
-      mUfs = UnderFileSystem.Factory
-          .create("/", UnderFileSystemConfiguration.defaults(ServerConfiguration.global()));
-    } else {
-      mUfs = UnderFileSystem.Factory.createForRoot(ServerConfiguration.global());
-    }
+    mRootUfsClient = masterContext.getUfsManager().getRoot();
+    mRootUfs = mRootUfsClient.acquireUfsResource();
```
Comment (calvinjia) [inline core/server/master/src/main/java/alluxio/master/meta/DefaultMetaMaster.java]: Does it make sense to keep these as variables, or could we acquire them when necessary?

Comment (apc999) [reply core/server/master/src/main/java/alluxio/master/meta/DefaultMetaMaster.java]: changed to acquire the resource when necessary. I think the code becomes more consistent and easier to reason.

**Comment by apc999** [comment]
@calvinjia  PTAL

**Comment by apc999** [comment]
ping @calvinjia

**Comment by calvinjia** [comment]
alluxio-bot, merge this please.

**Pull request merged**